Source/SessionDelegate.swift

```diff
@@ -193,7 +193,7 @@ extension SessionDelegate: URLSessionTaskDelegate {
     open func urlSession(_ session: URLSession, task: URLSessionTask, didCompleteWithError error: Error?) {
         eventMonitor?.urlSession(session, task: task, didCompleteWithError: error)
 
-        stateProvider?.request(for: task)?.didCompleteTask(task, with: error)
+        stateProvider?.request(for: task)?.didCompleteTask(task, with: error?.afError(or: .sessionTaskFailed(error: error!)))
```

Comment:
I don't think that's actually necessary. 
If the `error` variable is `nil`, the `afError(or:)` method is never called and the `error!` force unwrapping won't actually occur because it is wrapped in an `@autoclosure`, which won't be executed.

Reply:
Ah, you're right. To be safe, we could map the `Optional` first:
```swift
error.map { $0.asAFError(or: .sessionTaskFailed(error: $0)) }
```
A bit harder to read but we made it a point to try an remove as many IUOs from Alamofire as part of v5.

Reply:
Done.